nodedb.go

```diff
@@ -193,14 +213,23 @@ func (ndb *nodeDB) DeleteVersion(version int64, checkLatestVersion bool) {
 	ndb.deleteRoot(version, checkLatestVersion)
 }
 
+// DeleteVersion deletes a tree version from memory.
+func (ndb *nodeDB) DeleteMemoryVersion(version, previous int64, unsavedOrphans *map[string]int64) {
+	ndb.mtx.Lock()
+	defer ndb.mtx.Unlock()
+
+	ndb.deleteRoot(version, false)
+}
+
 // Saves orphaned nodes to disk under a special prefix.
 // version: the new version being saved.
 // orphans: the orphan nodes created since version-1
 func (ndb *nodeDB) SaveOrphans(version int64, orphans map[string]int64) {
 	ndb.mtx.Lock()
 	defer ndb.mtx.Unlock()
 
-	toVersion := ndb.getPreviousVersion(version)
+	// When saving a new version, the previous version will always be one less than the new version
```

Comment:
What happens when you save a version (H), delete it (H, so now the latest saved state is H-1), but then you save again?  Is it H+1 or H?